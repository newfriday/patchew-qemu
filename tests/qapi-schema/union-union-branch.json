{ 'enum': 'TestEnum',
  'data': [ 'value-a', 'value-b' ] }

{ 'enum': 'TestEnumA',
  'data': [ 'value-a1', 'value-a2' ] }

{ 'struct': 'TestTypeA1',
  'data': { 'integer': 'int' } }

{ 'struct': 'TestTypeA2',
  'data': { 'integer': 'int' } }

{ 'union': 'TestTypeA',
  'base': { 'type-a': 'TestEnumA' },
  'discriminator': 'type-a',
  'data': { 'value-a1': 'TestTypeA1',
            'value-a2': 'TestTypeA2' } }

{ 'struct': 'TestTypeB',
  'data': { 'integer': 'int' } }

{ 'union': 'TestUnion',
  'base': { 'type': 'TestEnum' },
  'discriminator': 'type',
  'data': { 'value-a': 'TestTypeA',
            'value-b': 'TestTypeB' } }
