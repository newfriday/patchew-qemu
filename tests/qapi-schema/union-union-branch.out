module ./builtin
object q_empty
enum QType
    prefix QTYPE
    member none
    member qnull
    member qnum
    member qstring
    member qdict
    member qlist
    member qbool
module union-union-branch.json
enum TestEnum
    member value-a
    member value-b
enum TestEnumA
    member value-a1
    member value-a2
object TestTypeA1
    member integer: int optional=False
object TestTypeA2
    member integer: int optional=False
object q_obj_TestTypeA-base
    member type-a: TestEnumA optional=False
object TestTypeA
    base q_obj_TestTypeA-base
    tag type-a
    case value-a1: TestTypeA1
    case value-a2: TestTypeA2
object TestTypeB
    member integer: int optional=False
object q_obj_TestUnion-base
    member type: TestEnum optional=False
object TestUnion
    base q_obj_TestUnion-base
    tag type
    case value-a: TestTypeA
    case value-b: TestTypeB
